File: credit/credit.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int test = 1;
int length(long card);
int sumdigit(long card,int len);
int main(void)
{
    long card = get_long("Enter card number: ");
    int x = length(card);
    if (!(x == 13 || x == 15 || x == 16))
    {
        printf("INVALID\n");
        test = 0;
    }
    int checksum = sumdigit(card, x);

}




int sumdigit(long card,int len)
{
    int sum = 0;
    if (len % 2 == 0)
    {
        for (int i = len-1; i > 1; i -= 1)
        {
            int card_odd = card
            sum += (card/pow(10,i));
            card %= pow(10,i)
        for (int j = len-2; j > 0; j -= 1)
        {
            sum += (card/pow(10,i));
        }
        }
    printf("%d",sum);
    }
    return sum;
}





int length(long card)
{
    int count = 0;
    while (card != 0)
    {
        card = card/10;
        count++;
    }
    return count;
}

File: sentimental-credit/credit.py
import cs50
import re
from math import ceil



def main():
    #Enter credit card number
    number = cs50.get_string("Enter credit card number: ")
    #check the length and return invalid if not correct length
    if not check_length(number) == 0:
        return
    sum_check(number)


#Function to check the length is 13, 15 or 16
def check_length(number):
    #regex check for length of string
    pattern = r"^[0-9]{13}$|^[0-9]{15}$|^[0-9]{16}$"
    if not re.match(pattern, number):
        print("INVALID")
        return 1
    return 0

# Sum check function to check card number is valid
def sum_check(number):
    sum = 0



    print(sum)


"""#checking the leading digits
def check_type(number):"""


main()